Emby.Server.Implementations/LiveTv/TunerHosts/HdHomerun/HdHomerunManager.cs

```diff
@@ -207,53 +220,70 @@ public async Task StartStreaming(IPAddress remoteIp, IPAddress localIp, int loca
         public async Task ChangeChannel(IHdHomerunChannelCommands commands, CancellationToken cancellationToken)
         {
             if (!_lockkey.HasValue)
+            {
                 return;
+            }
 
-            using (var tcpClient = _socketFactory.CreateTcpSocket(_remoteIp, HdHomeRunPort))
+            using (var tcpClient = new TcpClient(_remoteEndPoint))
+            using (var stream = tcpClient.GetStream())
             {
                 var commandList = commands.GetCommands();
-                var receiveBuffer = new byte[8192];
-
-                foreach (Tuple<string, string> command in commandList)
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(8192);
```

Comment:
What are the benefits of the new buffer allocation method? The new method uses a try-catch block which I generally try to avoid.

Reply:
It doesn’t allocate it gets it from an already allocated pool.